### PR TITLE
Add commands.json

### PR DESCRIPTION
**Pull request by thomascaudron**
Add `commands.json` with the overview of the different commands to be used in RedisInsight, Clients. redis.io and others.

### CODE REVIEW

File: commands.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
+{
+    "TFCALL": {
+        "summary": "Invoke a function with Triggers and Function",
```
Comment (MeirShpilraien) [inline commands.json:3]: `TFCALL` is not related to trigger, it just invoke a function ..

File: commands.json [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        "since": "2.0.0",
+        "group": "scripting",
+        "complexity": "Depends on the function that is executed.",
+        "arguments": [
+            {
+                "name": "function",
```
Comment (MeirShpilraien) [inline commands.json:9]: library.function ?

File: commands.json [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            {
+                "name": "arg",
+                "type": "string",
+                "optional": true,
+                "multiple": true
+            }
+        ]
+    },
+    "TFCALLASYNC": {
+        "summary": "Invoke a async function with Triggers and Function",
```
Comment (MeirShpilraien) [inline commands.json:32]: Same comment as above about triggers

File: commands.json [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        "since": "2.0.0",
+        "group": "scripting",
+        "complexity": "Depends on the function that is executed.",
+        "arguments": [
+            {
+                "name": "function",
```
Comment (MeirShpilraien) [inline commands.json:38]: Same comment as above

File: commands.json [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                    },
+                    {
+                        "name": "withcode",
+                        "type": "pure-token",
+                        "display_text": "withcode",
+                        "token": "WITHCODE",
+                        "optional": true
+                    },
+                    {
+                        "name": "verbose",
```
Comment (MeirShpilraien) [inline commands.json:127]: Can we somehow specify that this can be repeated ?

**Review by MeirShpilraien — COMMENTED** [review]
Same small comments.

File: commands.json [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                "summary": "Load a new library to Triggers and Functions",
+                "since": "2.0.0",
+                "group": "scripting",
+                "complexity": "O(1)",
+                "arguments": [
+                    {
+                        "name": "replace",
+                        "pure-token": "replace",
+                        "display_text": "replace",
+                        "token": "REPLACE",
```
[the remark this answers is not in the document]
Comment (thomascaudron) [reply commands.json:76]: The token is used for autocomplete, upper case can be used as is and lower case is to be replaced by user input.

File: commands.json [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+                ]
+            },
+            "TFUNCTION|DELETE": {
+                "summary": "Delete a library from Triggers and Functions by name",
+                "since": "2.0.0",
+                "group": "scripting",
+                "complexity": "O(1)",
+                "arguments": [
+                    {
+                        "name": "library-name",
```
[the remark this answers is not in the document]
Comment (thomascaudron) [reply commands.json:100]: That should be good, I took it from the [function-delete.json](https://github.com/redis/redis/blob/unstable/src/commands/function-delete.json)

File: commands.json (hunk 5 above)
Comment (thomascaudron) [reply commands.json:127]: do we have the same behaviour on another command? I couldn't find a pure-token type that had multiple in the Redis.

**Pull request closed without merging**